tests/hotplug/memory.go

```diff
@@ -89,12 +89,12 @@ var _ = Describe("[sig-compute][Serial]Memory Hotplug", decorators.SigCompute, d
 			return vm, vmi
 		}
 
-		getCurrentDomainMemory := func(vmi *v1.VirtualMachineInstance) *resource.Quantity {
+		getCurrentDomainMemory := func(vmi *v1.VirtualMachineInstance, g Gomega) *resource.Quantity {
 			domSpec, err := tests.GetRunningVMIDomainSpec(vmi)
-			ExpectWithOffset(1, err).NotTo(HaveOccurred())
-			ExpectWithOffset(1, domSpec.CurrentMemory).NotTo(BeNil())
+			g.ExpectWithOffset(1, err).NotTo(HaveOccurred())
+			g.ExpectWithOffset(1, domSpec.CurrentMemory).NotTo(BeNil())
 			memory, err := resource.ParseQuantity(fmt.Sprintf("%vKi", domSpec.CurrentMemory.Value))
-			ExpectWithOffset(1, err).ToNot(HaveOccurred())
+			g.ExpectWithOffset(1, err).ToNot(HaveOccurred())
 			return &memory
```

Comment:
my guess is that it's the `tests.GetRunningVMIDomainSpec()` execution that is occasionally flaky and probably not the parsing of the memory. 

If that's the case, then another option here would be to pass the domSpec into this function instead of the vmi, and move the `tests.GetRunningVMIDomainSpec()` to the top level Eventually loop.  That would prevent needing to pass `g Gomega` into the `getCurrentDomainMemory` function.

It's really not a big deal either way. Seeing the g Gomega being passed around outside of an Eventually or Consistently loop isn't common though.

Reply:
>my guess is that it's the tests.GetRunningVMIDomainSpec() execution that is occasionally flaky and probably not the parsing of the memory.

Correct. What I saw  is either incorrect pick of the launcher (the domain is still on source or already on target but we pick the other) or the second error was related to connecting to socket. Now I should have probably look at the `GetRunningVMIDomainSpec` but I admit I am being lazy here.

>If that's the case, then another option here would be to pass the domSpec into this function instead of the vmi, and move the tests.GetRunningVMIDomainSpec() to the top level Eventually loop. That would prevent needing to pass g Gomega into the getCurrentDomainMemory function.

That is indeed clearer for the reader, adopted.